src/content/docs/infrastructure/host-integrations/host-integrations-list/temporal-monitoring-integration.mdx

```diff
@@ -64,66 +68,131 @@ You can check the Temporal server running on the below urls:
-                         "workerCustomTag2",
-                         "workerCustomTag2Value"))
-            .reporter(reporter)
-            .reportEvery(com.uber.m3.util.Duration.ofSeconds(10));
-
-  // For Prometheus collection, expose the scrape endpoint at port 8077. See Micrometer documentation for details on starting the Prometheus scrape endpoint. For example,
-  HttpServer scrapeEndpoint = MetricsUtils.startPrometheusScrapeEndpoint(registry, 8077); //note: MetricsUtils is a utility file with the scrape endpoint configuration. See Micrometer docs for details on this configuration.
-  // Stopping the starter stops the HTTP server that exposes the scrape endpoint.
-  //Runtime.getRuntime().addShutdownHook(new Thread(() -> scrapeEndpoint.stop(1)));
-
-  //Create Workflow service stubs to connect to the Frontend Service.
-  WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(
-         WorkflowServiceStubsOptions.newBuilder()
-                 .setMetricsScope(scope) //set the metrics scope for the WorkflowServiceStubs
-                 .build());
+```java
+import com.sun.net.httpserver.HttpServer;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+
+public class MetricsUtils {
+   public static HttpServer startPrometheusScrapeEndpoint(
+           PrometheusMeterRegistry registry, int port) {
+       try {
+           HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+           server.createContext(
+                   "/metrics",
+                   httpExchange -> {
+                       String response = registry.scrape();
+                       httpExchange.sendResponseHeaders(200, response.getBytes(UTF_8).length);
+                       try (OutputStream os = httpExchange.getResponseBody()) {
+                           os.write(response.getBytes(UTF_8));
+                       }
+                   });
+
+           server.start();
+           return server;
+       } catch (IOException e) {
+           throw new RuntimeException(e);
+       }
+   }
+}
+```
+3. Add the blow dependency in build.gradle under the dependency section.
 
-  //Create a Workflow service client, which can be used to start, signal, and query Workflow Executions.
-  WorkflowClient yourClient = WorkflowClient.newInstance(service,
-         WorkflowClientOptions.newBuilder().build());
+```java
+implementation "io.micrometer:micrometer-registry-prometheus"
+```
+4. Update build.gradle with blow task  to start the worker.
 
-  //...
+```java
+task startMetricsWorker(type: JavaExec) {
+   mainClass = 'ProjectRoot.MetricsWorker'
+   classpath = sourceSets.main.runtimeClasspath
+}
 ```
-2. Go to your project directory and build.
+5. Go to your project directory and build.
 ```yml
 ./gradlew build
 ```
-3. Start the Worker.
+6. Start the Worker.
```

Comment:
```suggestion
6. Start the worker.
```

Reply:
Hi, @rhetoric101 , 
Thank you for your feedback. I've addressed your comments. Kindly review it.